### DISCUSSION TITLE
Use vscode treeview API

### DISCUSSION THREAD

**Discussion by shnitish** [microsoft/vscode-discussions#961] [Extension Development QnA] [answered]
Hi,

Currently I am working on a vscode extension for an internal project in my company. 
I wanted to implement a tree-like structure using the data from a DB.
I am already working on a view container of type "webview", I wanted to use the treeview API inside the current webview container without creating a new view in my sidebar.

I found nothing which claims that vscode has support for this, except creating a new view inside package.json with type "treeview" which basically creates a new view container below the existing webview.

I am not able to get any clues on this, can we do it or not, or is there any other way of doing it,

PS: already asked chatgpt, and based on its training data, it's not possible, so reaching out to community.

Thanks

**Comment by r3m0t** [comment]
Hi, you can create a view container with a webview and a treeview. The *view container* is the icon on the activity bar. It contains *views*. So, you don't have a view container of type webview - you have a view container with a single view, and that view is of type webview.

There's no way to put the treeview inside the webview.

**Comment by nitish-emids** [comment]
Hi, I am the same person who started this discussion from a different ID. 
Thanks for responding and I understood that there is no way to put the treeview inside the webview.
but I wonder how this treeview with tabs inside this panel is implemented, I wanted to do a similar thing but inside the side panel webview.

![image](https://github.com/microsoft/vscode-discussions/assets/143408852/0cb3aa69-ec5f-42b0-b769-2a4d5fb02c58)

**Reply by r3m0t** [reply, marked as answer]
The extensions cannot do everything that the built in features can do. Problems is not using the extension API for tree views.
